cachi2/core/package_managers/generic/main.py

```diff
@@ -102,3 +103,29 @@ def _load_lockfile(lockfile_path: RootedPath, output_dir: RootedPath) -> Generic
                 ),
             )
     return lockfile
+
+
+def _generate_sbom_components(lockfile: GenericLockfileV1) -> list[Component]:
+    """Generate a list of SBOM components for a given lockfile."""
+    components: list[Component] = []
+
+    for artifact in lockfile.artifacts:
+        name = Path(artifact.target).name
+        url = str(artifact.download_url)
+        checksums = ",".join([f"{algo}:{digest}" for algo, digest in artifact.checksums.items()])
```

Comment:
We're reporting all checksums present in the lockfile here, but this does not mean that all of them were validated. I'm wondering if this is the behavior we want. It'd be nice to double check how this is being done across different package managers, but I believe we can investigate this as a follow up, no need to block this PR.

Reply:
Huh, I guess that's technically right. I guess we would need to implement something like `must_match_all_checksums`? Good point though.

Reply:
From a quick look, it seems that the currently existing package managers are only reporting a single checksum in the SBOM. So I think it's worth to have a discussion about this before we end the experimental phase on this.

Reply:
> From a quick look, it seems that the currently existing package managers are only reporting a single checksum in the SBOM. So I think it's worth to have a discussion about this (should we report checksums that were not validated) before we end the experimental phase on generic.

@brunoapimentel can you please open an issue on this so that that discussion is not forgotten?